GridTariffApi.Lib.Tests/Services/V2/TariffPriceCacheTests.cs

```diff
@@ -45,12 +58,68 @@ public void TariffPersistenceCalledOnceTest()
         public void HolidayPersistenceCalledOnceTest()
         {
             Setup();
-            TariffPriceCache tariffPriceCache = new TariffPriceCache(_tariffPeristenceMock.Object, _holidayPeristenceMock.Object);
+            TariffPriceCache tariffPriceCache = new TariffPriceCache(_tariffPeristenceMock.Object, _holidayPeristenceMock.Object,null);
             for (int i = 0; i < 10; i++)
             {
                 tariffPriceCache.GetHolidays(DateTimeOffset.MinValue,DateTimeOffset.MaxValue);
             }
             _holidayPeristenceMock.Verify(x => x.GetHolidays(), Times.Once);
         }
+
+        [Fact]
+        public void InitMeteringPointIndexInitOnceTest()
+        {
+            Setup();
+            TariffPriceCache tariffPriceCache = new TariffPriceCache(_tariffPeristenceMock.Object, _holidayPeristenceMock.Object, _meteringPointPersistence.Object);
+
+            var reqParam = new List<String>();
+            reqParam.Add("mp1");
+            _meteringPointPersistence.Verify(x => x.GetMeteringPointsInformation(reqParam), Times.Never);
+            tariffPriceCache.GetMeteringPointInformation(reqParam);
+            _meteringPointPersistence.Verify(x => x.GetMeteringPointsInformation(reqParam), Times.Once);
+            tariffPriceCache.GetMeteringPointInformation(reqParam);
+            _meteringPointPersistence.Verify(x => x.GetMeteringPointsInformation(reqParam), Times.Once);
+        }
+
+        [Theory]
+        [InlineData("", "", "", "", 0)]
+        [InlineData("mp1", "", "", "", 1)]
+        [InlineData("mp1", "mp2", "", "", 2)]
+        [InlineData("mp1", "mp2", "mp3", "", 3)]
+        [InlineData("mp1", "mp2", "mp3", "mp4", 3)]
+        public void GetMeteringPointInformationTest(string firstId, string secondId, string thirdId, string fourthId, int numExpected)
+        {
+            Setup();
+            TariffPriceCache tariffPriceCache = new TariffPriceCache(_tariffPeristenceMock.Object, _holidayPeristenceMock.Object, _meteringPointPersistence.Object);
+
+            var param = new List<String>();
+            param.Add(firstId);
+            param.Add(secondId);
+            param.Add(thirdId);
+            param.Add(fourthId);
+            param.RemoveAll(a => a == "");
+
+            var retVal = tariffPriceCache.GetMeteringPointInformation(param);
+            Assert.Equal(numExpected, retVal.Count);
+
+            if (firstId != "")
```

Comment:
Kan iterere gjennom param-lista.

Reply:
Tar vekk denne dene testen, dekkes delvis av InitMeteringPointIndexInitOnceTest().
Bør lage ny test for MeteringPointPersistenceEF når vi er nærmere tilgjengelighet av makseffekt-verdier